### PR TITLE
Fix node tests ci

### PR DESCRIPTION
**Pull request by CarlSchwan**
- Update jest snapshot
- Fix importing 3rdparty vue components
- Disable temporarely tests using nextcloud-vue/Button

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	],
+    setupFilesAfterEnv: ['<rootDir>/tests/jestSetup.js'],
+	resetMocks: false,
+	collectCoverageFrom: [
+		'<rootDir>/apps/*/src/**/*.{js,vue}',
+	],
+	transform: {
+		// process `*.js` files with `babel-jest`
+		'.*\\.(js)$': 'babel-jest',
+		'src/utils/media/effects/virtual-background/vendor/*': 'jest-transform-stub',
```
Comment (PVince81) [inline jest.config.js:51]: we probably don't need that bit borrowed from the talk app ?

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	'@nextcloud/vue',
+	'splitpanes',
+	'string-length',
+	'strip-ansi',
+	'ansi-regex',
+	'char-regex',
+]
+
+module.exports = {
+	preset: '@vue/cli-plugin-unit-jest/presets/no-babel',
```
Comment (PVince81) [inline jest.config.js:37]: not sure if we can keep this row if you removed the external babel transformer

File: apps/dav/src/views/CalDavSettings.spec.js [hunk 3]
```diff
@@ -76,6 +76,7 @@ describe('CalDavSettings', () => {
 		)
 		expect(sendEventRemindersPush).toBeChecked()
 
+		return; // FIXME userEvent.click is broken with nextcloud-vue/Button
```
Comment (PVince81) [inline apps/dav/src/views/CalDavSettings.spec.js:79]:
```suggestion
		return // FIXME userEvent.click is broken with nextcloud-vue/Button
```

**Review by PVince81 — COMMENTED** [review]
Lint is failing

File: jest.config.js (hunk 1 above)
Comment (CarlSchwan) [reply jest.config.js:51]: Yep

File: jest.config.js (hunk 2 above)
Comment (CarlSchwan) [reply jest.config.js:37]: the external babel transformer that got removed was replaced by this preset, which allows us to use external vue components more easily

**Comment by MichaIng** [comment]
Looks like the new file needs to be added to tests to satisfy Drone checkers:
```
ERROR: There were 1 additional files:
jest.config.js
ERROR: Please remove or add those files again or inform the release team about those now files to be included or excluded from the release tar ball.```

File: apps/dav/src/views/CalDavSettings.spec.js [hunk 4]
```diff
@@ -76,6 +74,9 @@ describe('CalDavSettings', () => {
 		)
 		expect(sendEventRemindersPush).toBeChecked()
 
+		/*
+		FIXME userEvent.click is broken with nextcloud-vue/Button
```
Comment (PVince81) [inline apps/dav/src/views/CalDavSettings.spec.js:78]: link to ticket ?

File: jest.config.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// TODO: find a way to consolidate this in one place, with webpack.common.js
+const ignorePatterns = [
+	'vue-material-design-icons',
```
Comment (PVince81) [inline jest.config.js:25]: hope we can eventually centralize those as this is copied from the talk repo

and also wondering how many we still need now that IE / old Edge support is out

**Review by PVince81 — APPROVED** [review]
:+1: see comments

**Review by artonge — APPROVED** [review]

**Pull request merged**